src/js/components/Controls.js

```diff
@@ -44,6 +46,24 @@ class Controls extends React.PureComponent {
       if (this.mounted && this.state.showBorder === false) {
         this.setState({ showBorder: true });
       }
+
+      const currentScroll = this.$wrapper.offsetTop;
+
+      if(this.state.lastScrollPos > currentScroll){
+        this.setState({lastScrollPos: currentScroll});
+
+        if (!this.state.showControls) {
+          this.setState({showControls: true});
+        }
+      } else if( this.state.lastScrollPos < currentScroll) {
+        this.setState({lastScrollPos: currentScroll});
+
+        if (this.state.showControls &&
+           !this.props.showDisplayOptions &&
+           !this.props.showFontOptions) {
+          this.setState({showControls: false});
+        }
+      }
```

Comment:
Can you read state once and set it once?

```js
const { showDisplayOptions, showFontOptions } = this.props;

this.setState(prevState => {
 const { showControls, lastScrollPos } = prevState;

  if(lastScrollPos > currentScroll) {   
    return {
      lastScrollPos: currentScroll,
      showControls: !showControls
        ? true
        : showControls
    };
  } 

  return {
    lastScrollPos: currentScroll,
    showControls: showControls &&
      !showDisplayOptions &&
      !showFontOptions ?
        false :
        showControls
  };
});
```

Reply:
Thank you for this. Updated the code. Also switched to a flip animation instead of fade.